Api/Presentation/ExpensesController.cs

```diff
@@ -1,62 +1,160 @@
-using Api.Domain;
+using System.Security.Claims;
 
+using Api.Domain;
+using Api.Domain.Services;
 using AutoMapper;
-
-using Microsoft.AspNetCore.Authorization;
-
 using Microsoft.AspNetCore.Mvc;
 
 [ApiController]
 [Route("api/expenses")]
-[Authorize]
 public class ExpensesController : ControllerBase
 {
     private readonly IMapper _mapper;
+    private readonly IExpenseService _expenseService;
 
-    public ExpensesController(IMapper mapper)
+    public ExpensesController(IMapper mapper, IExpenseService expenseService)
     {
         _mapper = mapper;
+        _expenseService = expenseService;
+    }
+
+    [HttpPost]
+    public async Task<IActionResult> CreateExpense([FromBody] CreateExpenseRequest createExpenseRequest)
+    {
+        try
+        {
+            var userId = Guid.Parse(User.FindFirstValue(ClaimTypes.NameIdentifier));
+            if (createExpenseRequest == null)
+                return BadRequest("Invalid expense request.");
+
+            var expense = _mapper.Map<Expense>(createExpenseRequest);
+
+            expense.Id = Guid.NewGuid();
+            expense.CreatedAt = DateTime.UtcNow;
+
+            var success = await _expenseService.CreateAsync(expense);
+
+            if (success)
+                return CreatedAtAction(nameof(GetExpenseById), new { id = expense.Id }, _mapper.Map<ExpenseResponse>(expense));
+
+            return StatusCode(500, "An error occurred while creating the expense.");
+        }
+        catch (Exception e)
+        {
+            return StatusCode(500, $"Internal server error: {e.Message}");
+        }
     }
 
     [HttpGet]
     public async Task<IActionResult> GetAllExpenses()
     {
-        return Ok("expenses");
+        try
+        {
+            var expenses = await _expenseService.GetAllAsync();
+            return Ok(_mapper.Map<IEnumerable<ExpenseResponse>>(expenses));
+        }
+        catch (Exception e)
+        {
+            return StatusCode(500, $"Internal server error: {e.Message}");
+        }
     }
 
     [HttpGet("{id}")]
     public async Task<IActionResult> GetExpenseById(Guid id)
     {
-        return Ok("some expense");
+        try
+        {
+            var expense = await _expenseService.GetByIdAsync(id);
+            if (expense == null)
+                return NotFound();
+
+            return Ok(_mapper.Map<ExpenseResponse>(expense));
```

Comment:
add validation cases so that only the authenticated user can handle their own expenses like updating, deleting, etc. And not another user who is not the owner